libminifi/include/core/state/nodes/AgentInformation.h

```diff
@@ -425,77 +427,123 @@ class AgentStatus : public StateMonitorNode {
 
   std::vector<SerializedResponseNode> serialize() {
     std::vector<SerializedResponseNode> serialized;
+    auto serializedRepositories = serializeRepositories();
+    if (!serializedRepositories.empty()) {
+      serialized.push_back(serializedRepositories);
+    }
+    serialized.push_back(serializeUptime());
 
-    SerializedResponseNode uptime;
-
-    uptime.name = "uptime";
-    if (nullptr != monitor_) {
-      uptime.value = monitor_->getUptime();
-    } else {
-      uptime.value = "0";
+    auto serializedComponents = serializeComponents();
+    if (!serializedComponents.empty()) {
+      serialized.push_back(serializedComponents);
     }
 
-    if (!repositories_.empty()) {
-      SerializedResponseNode repositories;
+    serialized.push_back(serializeResourceConsumption());
 
-      repositories.name = "repositories";
+    return serialized;
+  }
+
+ protected:
+  SerializedResponseNode serializeRepositories() {
+    SerializedResponseNode repositories;
 
-      for (auto &repo : repositories_) {
-        SerializedResponseNode repoNode;
-        repoNode.collapsible = false;
-        repoNode.name = repo.first;
+    repositories.name = "repositories";
 
-        SerializedResponseNode queuesize;
-        queuesize.name = "size";
-        queuesize.value = repo.second->getRepoSize();
+    for (auto &repo : repositories_) {
```

Comment:
Does it modify anything on the original container?
I don't see such line, so I would go with const & here

Reply:
done [d2832cf](https://github.com/apache/nifi-minifi-cpp/pull/1032/commits/d2832cfcb3afa837e30e7bc95f496f0754d7603f)